### ISSUE TITLE
uBlock no longer works or appears after updating Firefox

### ISSUE THREAD

**Issue by Nawor3565** [gorhill/uBlock#3097]
Filter issues **MUST NOT** be reported here. Read first: <https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md>

### Describe the issue
After updating Firefox to version 56.0, the uBlock plugin appears to be totally non-functional, even after removing and reinstalling it. It does not show up as an add-on with the other add-ons I have, and doesn't block any ads. See screenshots below.

### One or more specific URLs where the issue occurs

N/A

### Screenshot in which the issue can be seen

![2017-10-05 1](https://user-images.githubusercontent.com/19783632/31252076-04dbc894-a9ee-11e7-83fd-2a4a0066c422.png)
![2017-10-05 2](https://user-images.githubusercontent.com/19783632/31252074-04d71f60-a9ee-11e7-86bf-85344682ee72.png)
![2017-10-05 3](https://user-images.githubusercontent.com/19783632/31252075-04d9258a-a9ee-11e7-923a-42dd0bba9c33.png)

### Steps for anyone to reproduce the issue

1: Update Firefox to version 56.0 from a previous version with uBlock Origin installed, uBlock will still be listed as an installed add-on but will have no effect.
2: Uninstall uBlock and reinstall from the official Firefox page. It will still be listed but again will have no effect.

### Your settings

[If you fail to provide this info, I will mark the issue as invalid. Lists all settings which differs from default settings]

- OS/version: Windows 10 update 1703
- Browser/version: Firefox 56.0
- uBlock Origin version: 1.14.0

##### Your filter lists
N/A, cannot access them anyway.

##### Your custom filters (if any)
N/A

**Comment by gorhill** [comment]
Unable to repro.

Please investigate more thoroughly your issue to make the case the issue is with uBO.

For instance you should have look at browser console and also try installing only uBO in a new profile.

I will re-open if you make the case the issue is really with uBO. I am also assuming you read carefully the release notes. And especially, read CONTRIBUTING.

**Comment by Nawor3565** [comment]
Alright, I read through the docs you mentioned and did some more digging. It looks like this only is happening with uBlock versions 1.13.10 and above, because version 1.13.8 is working fine. 1.13.10 is the first version with webext, and it seems that it might be related. Disabling all my legacy plugins didn't make a difference. Next I'll make a new profile and install FF and uBO fresh and update the issue.

**Comment by Nawor3565** [comment]
I used a new Windows profile all together, with a newly "refreshed" FF 56.0 and no other add-ons, and uBO 1.14.10 installed and worked fine. Here's the browser log for installing the add-on:
`1507241481954   addons.xpi  WARN    Please specify whether you want browser_style or not in your options_ui options. 
1507241484743   addons.xpi  WARN    Failed to remove temporary file C:\Users\extra\AppData\Local\Temp\tmp-zys.xpi for addon 
https://addons.mozilla.org/firefox/downloads/latest/ublock-origin/addon-607454-latest.xpi?src=dp-btn-primary: [Exception... "Component returned failure code: 0x80520015 (NS_ERROR_FILE_ACCESS_DENIED) [nsIFile.remove]"  nsresult: "0x80520015 (NS_ERROR_FILE_ACCESS_DENIED)"  location: "JS frame :: resource://gre/modules/addons/XPIInstall.jsm :: removeTemporaryFile :: line 1596"  data: no] Stack trace: removeTemporaryFile()@resource://gre/modules/addons/XPIInstall.jsm:1596 < startInstall/<()@resource://gre/modules/addons/XPIInstall.jsm:1964`
I went back to my normal FF where I'm having the problem, and tried installing uBO again. Here's the browser log:
`1507241678952   addons.xpi  WARN    Please specify whether you want browser_style or not in your options_ui options. 
1507241680448   addons.xpi  WARN    Exception running bootstrap method startup on uBlock0@raymondhill.net: Error: Out-of-process WebExtensions are not supported with multiple child processes (resource://gre/modules/Extension.jsm:788:13) JS Stack trace: this.Extension<@Extension.jsm:788:13 < startup@WebExtensionBootstrap.js:28:15 < callBootstrapMethod@XPIProvider.jsm:4405:11 < startInstall/<@XPIInstall.jsm:1935:13 
1507241680450   addons.xpi  WARN    Failed to remove temporary file C:\Users\Nawor\AppData\Local\Temp\tmp-7yq.xpi for addon https://addons.mozilla.org/firefox/downloads/latest/ublock-origin/addon-607454-latest.xpi?src=dp-btn-primary: [Exception... "Component returned failure code: 0x80520015 (NS_ERROR_FILE_ACCESS_DENIED) [nsIFile.remove]"  nsresult: "0x80520015 (NS_ERROR_FILE_ACCESS_DENIED)"  location: "JS frame :: resource://gre/modules/addons/XPIInstall.jsm :: removeTemporaryFile :: line 1596"  data: no] Stack trace: removeTemporaryFile()@resource://gre/modules/addons/XPIInstall.jsm:1596 < startInstall/<()@resource://gre/modules/addons/XPIInstall.jsm:1964`
It looks like the "Out-of-process WebExtensions are not supported with multiple child processes" is the problem, but I don't know enough about add-ons to know how to fix it. I might try copying my FF profile and removing all the add-ons I have later tonight.

**Comment by gorhill** [comment]
> It looks like the "Out-of-process WebExtensions are not supported with multiple child processes" is the problem

I am using uBO with OOP WebExtensions -- `extensions.webextensions.remote` is `true` -- there is no issue on my side. Probably should look in `about:support` to find out which `about:config` you modified from default values and see what in there could be the issue.

**Comment by Nawor3565** [comment]
Took a look in about:support, there's a bunch of stuff there but it seems to all be normal Firefox settings, I'll paste it [here](https://pastebin.com/Ktmr7EgR) just in case it comes in handy. I asked about the Out-of-process WebExt error on the official FF support forums, I'll also post whatever I find out there.

**Comment by gwarser** [comment]
Reset `dom.ipc.processCount.extension` to `1`

**Comment by Nawor3565** [comment]
Alright, I managed to fix it! I still have no idea what caused the issue in the first place, but the [answer](https://support.mozilla.org/en-US/questions/1178555) I got on my forum question helped me fix it. I just copied the profile files to a new one like the article mentions, so I didn't lose very much data, but when I signed into my Firefox Sync account it started syncing my other addons before I could tell it not to. So my other addons and data are all fine and everything seems to be working.
